marimo/_save/ast.py

```diff
@@ -106,31 +112,69 @@ def generic_visit(self, node: ast.AST) -> tuple[ast.Module, ast.Module]:  #  typ
         # excluding by omission try, for, classes and functions.
         if len(on_line) == 0:
             if isinstance(previous, (ast.With, ast.If)):
-                try:
-                    # Recursion by referring the the containing block also
-                    # captures the case where the target line number was not
-                    # exactly hit.
-                    return ExtractWithBlock(self.target_line).generic_visit(
-                        previous.body  # type: ignore[arg-type]
-                    )
-                except BlockException:
-                    on_line.append(previous)
+                on_line.append(previous)
+                bodies = (
+                    [previous.body]
+                    if isinstance(previous, ast.With)
+                    else [previous.body, previous.orelse]
+                )
+                for body in bodies:
+                    try:
+                        # Recursion by referring the the containing block also
+                        # captures the case where the target line number was not
+                        # exactly hit.
+                        return ExtractWithBlock(
+                            self.target_line
+                        ).generic_visit(
+                            body  # type: ignore[arg-type]
+                        )
+                    except BlockException:
+                        pass
```

Comment:
I know this logic mostly existed before, but can you also add an example here that illustrates the kind of code that would trigger this code path? What is an example of a block that is contained with another?

Also, is this visitor used for both `With` and `If` blocks, or just `With` as the name implies?

Reply:
Yeah, the logic gets a little weird here. Tried to add in some more comments as best as I remembered it